src/modules/widgets/widgetsSaga.ts

```diff
@@ -321,6 +323,87 @@ export function* cloneWidget({
   yield put(saveDashboard(dashboardId));
 }
 
+export function* createNewChart({
+  payload,
+}: ReturnType<typeof createNewChartAction>) {
+  const { widgetId } = payload;
+
+  yield put(appActions.showQueryPicker());
+  const action = yield take([
+    SELECT_SAVED_QUERY,
+    CREATE_QUERY,
+    appActions.hideQueryPicker.type,
+  ]);
+
+  if (action.type === appActions.hideQueryPicker.type) {
```

Comment:
Is there any possibility to extract common logic for two flows:
- Select Saved Query for widget
- Create new Query for widget

Then re-use it for two cases:
1. User drops chart widget on Grid
2. User decides to re-connect query after it was removed 

Currently I think that the `createNewChart` saga flow - creates a lot of code duplication.